### PR TITLE
Improvements to Documentation

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #583

### CODE REVIEW

File: website/docs/plugin/framework/handling-data/attributes.mdx [hunk 1]
```diff
@@ -468,6 +468,308 @@ attribute value in a way that Terraform will understand. `attr.Value` has the
...
-| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `ToTerraformValue` | Returns a Go type that is valid input for [`tftypes.NewValue`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-go/tftypes#NewValue) for the `tftypes.Type` specified by the `attr.Type` that creates the `attr.Value`. |
 | `Equal`            | Returns true if the passed attribute value should be considered to the attribute value the method is being called on. The passed attribute value is not guaranteed to be of the same Go type.                                   |
+
+## Terraform Configuration and Schema
+
+### Simple Attributes
```
Comment (bflad) [inline website/docs/plugin/framework/handling-data/attributes.mdx:477]: The "Framework Types and Value Types" above walks through many of these Simple/Collection configuration and schema examples although it looks like they need to be updated for `schema.XXXAttribute` syntax (which I'll handle separately).

File: website/docs/plugin/framework/handling-data/attributes.mdx [hunk 2]
```diff
@@ -468,6 +468,308 @@ attribute value in a way that Terraform will understand. `attr.Value` has the
...
+                    "set_attribute":     types.ListType{ElemType: types.StringType},
+                    "string_attribute":  types.StringType,
+                },
+            },
+        },
+    }
+}
+```
+
+### Nested Attributes
```
Comment (bflad) [inline website/docs/plugin/framework/handling-data/attributes.mdx:626]: I think these are omitted from the above sections, so it'll be good to walk through these. Do you think you could structure this new content to look more similar to the existing content (e.g. split per attribute type). Thanks!

**Review by bflad — COMMENTED** [review]
Nice!

File: website/docs/plugin/framework/handling-data/attributes.mdx (hunk 2 above)
Comment (bflad) [reply website/docs/plugin/framework/handling-data/attributes.mdx:626]: Ah ha, it was awkwardly on the Schemas page -- I think that content can be moved here. 👍

File: website/docs/plugin/framework/handling-data/attributes.mdx (hunk 1 above)
Comment (bendbennett) [reply website/docs/plugin/framework/handling-data/attributes.mdx:477]: Ok. I'll remove these, I just thought there might be some value in being able to quickly scan the TF configuration and the schema to see the correlation for simple attributes between the two.

File: website/docs/plugin/framework/handling-data/attributes.mdx (hunk 2 above)
Comment (bendbennett) [reply website/docs/plugin/framework/handling-data/attributes.mdx:626]: Have moved the content and consolidated the TF configuration and schema examples for nested attributes.

**Review by bflad — APPROVED** [review]
Looks great to me 🚀

**Pull request merged**